services/OrderService/Repository/OrderItemRepository.cs

```diff
@@ -1,37 +1,54 @@
+using Microsoft.EntityFrameworkCore;
+using OrderService.Data;
 using OrderService.Models;
 using OrderService.Repository;
 
 namespace OrderService.Models;
 
 public class OrderItemRepository : IOrderItemRepository
 {
-    public Task<OrderItem> CreateAsync(OrderItem orderItemModel)
+
+    private readonly InMemoryContext _context;
+
+    public OrderItemRepository(InMemoryContext context)
+    {
+        _context = context;
+    }
+
+    public async Task CreateAsync(OrderItem orderItemModel)
     {
-        throw new NotImplementedException();
+        await _context.OrderItems.AddAsync(orderItemModel);
     }
 
-    public Task<OrderItem> DeleteAsync(int id)
+    public async Task DeleteAsync(OrderItem orderItemModel)
     {
-        throw new NotImplementedException();
+        _context.Remove(orderItemModel);
     }
 
-    public Task<bool> Exists(int id)
+    public async Task<bool> Exists(int id)
     {
-        throw new NotImplementedException();
+        var order = await _context.OrderItems.FirstAsync(x => x.Id == id);
+        return order == null ? true : false;
```

Comment:
it should be return order == null ? false : true; or order != null ? true : false. This is returning true if it's null which is the opposite it should do.